app/models/krikri/field_value_report.rb

```diff
@@ -0,0 +1,167 @@
+    # represents data about a document.
+    #   @example: [{ :id=>"a1b2c3",
+    #                :sourceResource_title=>"Moomin Story",
+    #                :isShownAt_id=> "http://example.org/items/4321" }]
+    #
+    # @!attribue :provider_id [String]
+    attr_accessor :name, :data, :provider_id
+
+    ##
+    # @param [String] name represents the name of the field value report, ie.
+    # 'sourceResource_title'
+    #
+    # @param [String] provider_id represents the id of the provider.
+    #
+    # The two params act as a compound key.
+    #
+    # @return [Krikri::FieldValueReport]
+    def self.find(name, provider_id)
+      return nil unless fields.include? name.to_sym
+      provider = Krikri::Provider.find(provider_id)
+      return nil unless provider.present?
+      new({ :name => name,
+            :provider_id => provider_id,
+            :data => compile_data(name, provider) })
+    end
+
+    ##
+    # All of the fields for which a report can be created.
+    def self.fields
+      [:dataProvider_providedLabel,
+       :sourceResource_alternative_providedLabel,
+       :sourceResource_collection_title,
+       :sourceResource_contributor_providedLabel,
+       :sourceResource_creator_providedLabel,
+       :sourceResource_date_providedLabel,
+       :sourceResource_description,
+       :sourceResource_format,
+       :sourceResource_genre_providedLabel,
+       :sourceResource_language_providedLabel,
+       :sourceResource_publisher_providedLabel,
+       :sourceResource_rights,
+       :sourceResource_rightsHolder_providedLabel,
+       :sourceResource_spatial_providedLabel,
+       :sourceResource_subject_providedLabel,
+       :sourceResource_temporal,
+       :sourceResource_title,
+       :sourceResource_type_name]
+    end
+
+    def data
+      @data ||= initialize_data
+    end
+
+    ##
+    # Gets the provider associated with @provider_id
+    #
+    # @return [Krikri::Provider]
+    def provider
+      return nil unless provider_id.present?
+      Krikri::Provider.find(provider_id)
+    end
+
+    ##
+    # The headers for the report table.  Each row (ie. each Hash in the @data
+    # Array) is expected to have a single value for each header.
+    def headers
+      [:id, name.to_sym, :isShownAt_id]
+    end
+
+    ##
+    # Get a csv table representation of data.
+    #
+    # @return [CSV::Table]
+    def csv
+      table = CSV::Table.new([CSV::Row.new(headers, [], true)])
+      return table unless data.present?
+
+      rows = data.map do |row|
+        values = headers.map { |header| row[header] }
+        CSV::Row.new(headers, values)
+      end
+
+      rows.each { |r| table << r }
+      table
+    end
+
+    ##
+    # @return Array[<Hash>]
+    def self.compile_data(name, provider)
+      data = []
+      query_params = { :fq => "provider_id:\"#{provider.rdf_subject}\"",
+                       :fl => ['id', 'isShownAt_id', name],
+                       :rows => 1000,
```

Comment:
1000 rows at a time may not be optimal - we will have to experiment with larger data sets.  If anyone has a good estimation of what the desired number of rows at a time would be, please let me know.  1000 is a blind guess.